### PR TITLE
Hotfix issue 399

### LINKED ISSUE #399: Cannot checkout tutorials develop in tests
@valentin-seitz I guess I am doing something wrong, but this is important to be able to test all develop branches together.

Executing

```shell
gh workflow run run_testsuite_manual.yml -f suites=fenics_test -f build_args="PRECICE_REF:develop,OPENFOAM_ADAPTER_REF:develop,PYTHON_BINDINGS_REF:develop,FENICS_ADAPTER_REF:develop,TUTORIALS_REF:develop" -f loglevel=DEBUG --ref=develop
```

[results](https://github.com/precice/tutorials/actions/runs/6922805250/job/18829951701) into:

```
DEBUG: Checking out tutorials develop before copying
fatal: Refusing to fetch into current branch refs/heads/develop of non-bare repository
Traceback (most recent call last):
  File "/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials/tools/tests/systemtests/Systemtest.py", line 234, in _fetch_ref
    result = subprocess.run([
             ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 569, in run
Using log-level: DEBUG
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', '-C', '/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials', 'fetch', 'origin', 'develop:develop']' returned non-zero exit status 128.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials/tools/tests/systemtests.py", line 96, in <module>
    main()
  File "/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials/tools/tests/systemtests.py", line 75, in main
    result = systemtest.run(run_directory)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials/tools/tests/systemtests/Systemtest.py", line 490, in run
    self.__prepare_for_run(run_directory)
  File "/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials/tools/tests/systemtests/Systemtest.py", line 479, in __prepare_for_run
    self.__copy_tutorial_into_directory(run_directory)
  File "/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials/tools/tests/systemtests/Systemtest.py", line 271, in __copy_tutorial_into_directory
    self._fetch_ref(PRECICE_TUTORIAL_DIR, ref_requested)
  File "/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials/tools/tests/systemtests/Systemtest.py", line 245, in _fetch_ref
    raise RuntimeError(f"An error occurred while fetching origin '{ref}':  {e}")
RuntimeError: An error occurred while fetching origin 'develop':  Command '['git', '-C', '/home/precice/runners_root/actions-runner-tutorial/_work/tutorials/tutorials', 'fetch', 'origin', 'develop:develop']' returned non-zero exit status 128.
```

[Similarly](https://github.com/precice/tutorials/actions/runs/6922823475/job/18829990141) when using a specific commit.

### PR DESCRIPTION
**Pull request by valentin-seitz**
Executing `gh workflow run run_testsuite_manual.yml -f suites=fenics_test -f build_args="TUTORIALS_REF:hotfix-issue-399" -f loglevel=DEBUG 
-f systests_branch=hotfix-issue-399 --ref=hotfix-issue-399` [now works](https://github.com/precice/tutorials/actions/runs/6935990599). 
This should fix #399

### CODE REVIEW

File: .github/workflows/run_testsuite_workflow.yml [hunk 1]
```diff
@@ -49,6 +49,9 @@ jobs:
         ref: ${{ inputs.systests_branch }}
         lfs: true
         fetch-depth: 0
+    - name: Map remote branches to local ones
+      run: |
+        /home/precice/runners_root/scripts/map_origin_to_local.sh ./
```
Comment (MakisH) [inline .github/workflows/run_testsuite_workflow.yml:54]: Shouldn't this script also be stored in this or at least in the infrastructure repository?

**Review by MakisH — APPROVED** [review]
Seems to work, at least the error is not there anymore, but I am not sure how to check that it is indeed checking out the right branch/commit.

File: .github/workflows/run_testsuite_workflow.yml (hunk 1 above)
Comment (valentin-seitz) [reply .github/workflows/run_testsuite_workflow.yml:54]: I can put it into the Infrastructure repo. I also got some input on how i could resolve this issue without it beeing this ugly, but its more a hotfix for now. We def. should address this in the python part.

**Comment by valentin-seitz** [comment]
> Seems to work, at least the error is not there anymore, but I am not sure how to check that it is indeed checking out the right branch/commit.

Since we basically restore a certain path from a given ref via the `git checkout` its not really easy to check it. 
But if the `git checkout` command itself fails, python will throw an exception. 
So: It should work or it will generate an Exception (which are still a bit hard to debug because we dont get the orignal stderr/stdout of the git subprocess).

File: .github/workflows/run_testsuite_workflow.yml (hunk 1 above)
Comment (MakisH) [reply .github/workflows/run_testsuite_workflow.yml:54]: There is nothing more permanent than a temporary workaround! :smile: 

Feel free to merge once the script is uploaded somewhere.

**Pull request merged**